contracts/integrations/examples/contractCreator/withoutTreasury/TimeTaskCreator.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "../../../OpsTaskCreator.sol";
+
+/**
+ * @dev
+ * Example contract that creates a time task.
+ */
+// solhint-disable not-rely-on-time
+// solhint-disable no-empty-blocks
+contract TimeTaskCreator is OpsTaskCreator {
+    uint256 public count;
+    uint256 public lastExecuted;
+    bytes32 public taskId;
+    uint256 public constant MAX_COUNT = 5;
+    uint256 public constant INTERVAL = 3 minutes;
+
+    event CounterTaskCreated(bytes32 taskId);
+
+    constructor(address payable _ops, address _owner)
+        OpsTaskCreator(_ops, _owner)
+    {}
+
+    function createTask() external payable {
+        require(taskId == bytes32(""), "Already started task");
+
+        bytes memory execData = abi.encodeCall(this.increaseCount, (1));
+
+        ModuleData memory moduleData = ModuleData({
+            modules: new Module[](2),
+            args: new bytes[](2)
+        });
+        moduleData.modules[0] = Module.TIME;
+        moduleData.modules[1] = Module.PROXY;
+
+        moduleData.args[0] = _timeModuleArg(block.timestamp, INTERVAL);
+        moduleData.args[1] = _proxyModuleArg();
+
+        bytes32 id = ops.createTask(
+            address(this),
+            execData,
+            moduleData,
+            address(0)
+        );
+
+        taskId = id;
+        emit CounterTaskCreated(id);
+    }
+
+    function increaseCount(uint256 _amount) external onlyDedicatedMsgSender {
+        uint256 newCount = count + _amount;
+
+        if (newCount >= MAX_COUNT) {
+            ops.cancelTask(taskId);
+            count = 0;
+        } else {
+            count += _amount;
+            lastExecuted = block.timestamp;
+        }
+
+        (uint256 amount, address feeToken) = _getFeeDetails();
+
+        _transfer(amount, feeToken);
```

Comment:
should we have a method that do both `_payGelatoFee()` ?

Reply:
I think some users would benefit from knowing `amount` & `feeToken` e.g. they would need to invoke another contract to get the funds before paying Gelato.

I could add an additional method `_payGelatoFee()` that does both too. What do u think.

Reply:
good point, let's keep it like that